solr/core/src/java/org/apache/solr/api/V2HttpCall.java

```diff
@@ -360,6 +366,85 @@ protected void execute(SolrQueryResponse rsp) {
     SolrCore.postDecorateResponse(handler, solrReq, rsp);
   }
 
+  @Override
+  protected void populateTracingSpan(Span span) {
+    // Set db.instance
+    String coreOrColName = this.origCorename;
+    if (coreOrColName == null) {
+      Map<String, String> pathTemplateValues = getUrlParts(); // == solrReq.getPathTemplateValues()
+      coreOrColName = pathTemplateValues.get("collection");
+      if (coreOrColName == null) {
+        coreOrColName = pathTemplateValues.get("core");
+      }
+    }
+    if (coreOrColName != null) {
+      span.setTag(Tags.DB_INSTANCE, coreOrColName);
+    }
+
+    // Get the templatize-ed path
+    String path;
+    if (api instanceof AnnotatedApi) {
+      // ideal scenario; eventually everything might be AnnotatedApi?
+      var aapi = (AnnotatedApi) api;
+      path = aapi.getEndPoint().path()[0]; // consider first to be primary
+    } else {
+      path = computeEndpointPath();
+      // TODO consider getValidators, looking for command & path?
+    }
+
+    String verb = req.getMethod().toLowerCase(Locale.ROOT);
+    // if this api has commands ...
+    final Map<String, JsonSchemaValidator> validators = getValidators(); // should be cached
+    if (validators != null && validators.isEmpty() == false && solrReq != null) {
+      // does this request have one command?
+      List<CommandOperation> cmds = solrReq.getCommands(false);
+      if (cmds.size() == 1) {
+        verb = cmds.get(0).name;
+      }
+    }
+
+    span.setOperationName(verb + ":" + path);
+
+    if (req.getQueryString() != null) {
+      span.setTag("params", req.getQueryString());
+    }
```

Comment:
I did a change here that is debatable.  Previously, this was the SolrParams toString but is now the non-decoded / raw query string (thus has percent-encoding).  But I think I want to change this back.  This is all guarded by no-op, and furthermore we could modify SolrQueryRequestBase to cache the toString of the original params if we want.  That method is also called by SolrCore.preDecorateResponse for logging.